packages/interactivity/src/utils.ts

```diff
@@ -48,12 +56,18 @@ const afterNextFrame = ( callback: () => void ) => {
  *
  * @return Promise
  */
-export const splitTask = () => {
-	return new Promise( ( resolve ) => {
-		// TODO: Use scheduler.yield() when available.
-		setTimeout( resolve, 0 );
-	} );
-};
+export const splitTask =
+	'scheduler' in window &&
+	typeof window.scheduler === 'object' &&
+	null !== window.scheduler &&
+	'yield' in window.scheduler &&
+	typeof window.scheduler.yield === 'function'
+		? window.scheduler.yield
+		: () => {
+				return new Promise( async ( resolve ) => {
```

Comment:
No need for `async` here.

```suggestion
				return new Promise( ( resolve ) => {
```